bitnami/prometheus-operator/templates/alertmanager/service.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if .Values.alertmanager.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "prometheus-operator.alertmanager.fullname" . }}
+  labels: {{ include "prometheus-operator.alertmanager.labels" . | nindent 4 }}
+  {{- with .Values.alertmanager.service.annotations }}
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  type: {{ .Values.alertmanager.service.type }}
+  {{- if and .Values.alertmanager.service.loadBalancerIP (eq .Values.alertmanager.service.type "LoadBalancer") }}
+  loadBalancerIP: {{ .Values.alertmanager.service.loadBalancerIP }}
+  {{- end }}
+  {{- if and (eq .Values.alertmanager.service.type "LoadBalancer") .Values.alertmanager.service.loadBalancerSourceRanges }}
+  {{- with .Values.alertmanager.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  {{- end }}
+  {{- if and (eq .Values.alertmanager.service.type "ClusterIP") .Values.alertmanager.service.clusterIP }}
+  clusterIP: {{ .Values.alertmanager.service.clusterIP }}
+  {{- end }}
+  ports:
+  - name: http
+    port: {{ .Values.alertmanager.service.port }}
+    targetPort: 9093
+    {{- if and .Values.alertmanager.service.nodePort (or (eq .Values.alertmanager.service.type "NodePort") (eq .Values.alertmanager.service.type "LoadBalancer")) }}
+    nodePort: {{ .Values.alertmanager.service.nodePort }}
+    {{- end }}
+  selector:
+    app: alertmanager
```

Comment:
Instead of `app`, you could use `app.kubernetes.io/name`

Reply:
Wait, I see that you use it. Not sure about the role of `app: alertmanager` here.

Reply:
These labels are added by the `prometheus-operator` component. We have to use whatever labels they've defined in the code

ref: https://github.com/coreos/prometheus-operator/blob/master/pkg/alertmanager/statefulset.go#L195-L197